sky/skylet/log_lib.py

```diff
@@ -440,15 +476,39 @@ def tail_logs(job_id: Optional[int],
         with open(log_path, 'r', newline='', encoding='utf-8') as log_file:
             # Using `_follow` instead of `tail -f` to streaming the whole
             # log and creating a new process for tail.
+            if tail > 0:
+                current_file_lines = log_file.readlines()
+                lines = collections.deque(current_file_lines, maxlen=tail)
+                start_stream = _should_start_streaming(current_file_lines,
+                                                       lines, start_stream_at)
+                for line in lines:
+                    if start_stream_at in line:
+                        start_stream = True
+                    if start_stream:
+                        print(line, end='')
+                # Flush the last n lines
+                print(end='', flush=True)
+            # Now, the cursor is at the end of the last lines
+            # if tail > 0
```

Comment:
I found this over complicated a bit. Since we are already processing the all the lines of the file, why can't we just do so by something like the following:
```
def _print_lines_after_start_stream_at(log_content, start_stream_at, tail) -> bool:
    for line_idx, line in log_content[:PEEK_HEAD_LINES_FOR_START_STREAM]:
        if start_stream_at in line:
            break
    else:
        return False
    tail_start_idx = len(log_content) - tail
    start_stream = tail_start_idx >= line_idx
    log_content = log_content[-tail:]
    for line in log_content:
        if start_stream_at in line:
            start_stream = True
        if start_stream:
            print(line, end='')
    print(end='', flusth=True)
```

```suggestion
            if tail > 0:
                start_stream = _print_lines_after_start_stream_at(log_file.readlines(), start_stream_at, tail)
            # Now, the cursor is at the end of the last lines
            # if tail > 0
```

Reply:
Since we can't load the whole file into memory, I changed the approach but can't resolve this comment.